fs/grub/disk_io.c

```diff
@@ -125,36 +125,30 @@ int filepos;
 int filemax;
 
 //Cromwell hooks
-int
-rawread (int drive, int sector, int byte_offset, int byte_len, char *buf)
+int rawread(int drive, int sector, int byte_offset, int byte_len, char *buf)
 {
-	sector+=byte_offset/512;
-	byte_offset%=512;
-	
-	while(byte_len) {
-		int nThisTime=512;
-	       	if(byte_len<512) nThisTime=byte_len;
-		if(byte_offset) {
-			u8 ba[512];
-			if(BootIdeReadSector(drive, buf, sector, 0, 512)) {
-				bprintf("Unable to get first sector\n");
-                             	return 1;
-			}
-			memcpy(buf, &ba[byte_offset], nThisTime-byte_offset);
-                      	buf+=nThisTime-byte_offset;
-                   	byte_len-=nThisTime-byte_offset;
-                      	byte_offset=0;
-             	} else {
-                     	if(BootIdeReadSector(drive, buf, sector, 0, nThisTime)) {
-                      		bprintf("Unable to get first sector\n");
-                             	return 1;
-                     	}
-                   	buf+=nThisTime;
-                   	byte_len-=nThisTime;
-              	}
+	// Advance sector counter by offset to get sub-sector-size offset
```

Comment:
nit: A comment that we assume that `byte_offset >= 0` would be good.

If this codebase uses `assert` (or similar), use that.

Reply:
Improved as well.